bin/bovine.py

```diff
@@ -15,7 +15,7 @@ def __init__(self, root_directory):
     # import the yaml from the files that we found above
```

Comment:
using self.filenames = filenames in the __init__ method doesn't make any sense.  Within the scope of this method, there is no var called filenames.  This is where you current error is.